tex/generic/pgf/libraries/shapes/pgflibraryshapes.geometric.code.tex

```diff
@@ -210,10 +210,15 @@
 %
 \pgfkeys{/pgf/.cd,
     aspect/.code={\pgfsetshapeaspect{#1}},% this for tikz...
-    shape aspect/.initial=1,% but this is consistent with other pgfset stuff.
+    shape aspect/.initial=none,% but this is consistent with other pgfset stuff.
     shape aspect/.code={%
-        \pgfkeys{/pgf/aspect=#1}%
-        \pgfkeyssetvalue{/pgf/shape aspect}{#1}
+        \edef\pgf@lib@temp{#1}%
+        \ifx\pgf@lib@temp\pgf@nonetext
+          \pgfkeys{/pgf/aspect=1}%
+        \else
+          \pgfkeys{/pgf/aspect=#1}%
+        \fi
+        \pgfkeyssetvalue{/pgf/shape aspect}{#1}%
```

Comment:
If the default was empty the code could be simplified (untested).
```suggestion
        \pgfqkeys{/pgf}{aspect/.expanded=\pgfutil@ifxempty\pgf@lib@temp{1}{#1}}%
        \pgfkeyssetvalue{/pgf/shape aspect}{#1}%
```
Same for the other conditionals.